### DISCUSSION TITLE
Failed to fetch log file from worker. Server disconnected without sending a response.

### DISCUSSION THREAD

**Discussion by joeyyuan88** [apache/airflow#21583] [General]
### Apache Airflow version

2.2.3 (latest released)

### What happened

please help, thanks

use CeleryExecutor with multiple workers on different servers

*** Log file does not exist: /opt/airflow/logs/example_python/print_the_context/2022-02-08T07:08:16.294434+00:00/1.log
*** Fetching from: http://phxfb-spias-u10-worker3:8793/log/example_python/print_the_context/2022-02-08T07:08:16.294434+00:00/1.log
*** Failed to fetch log file from worker. Server disconnected without sending a response.

phxfb-spias-u10-worker3 has configured to right ip in /etc/hosts

when I manually call the api inside docker container got 403

default@server-worker1:/opt/airflow$ curl http://localhost:8793/log
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>403 Forbidden</title>
<h1>Forbidden</h1>
<p>You don't have the permission to access the requested resource. It is either read-protected or not readable by the server.</p>

### Operating System

apache/airflow:2.2.3-python3.8

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
You cannot read the logs without authentication - that's why you get forbidden (you need to pase secret key as authentication to read the logs this way). 

However your problem is likely connected with your deployment. You need to make sure that there are no firewals blocking and that your networking allow your webserver to communicate with the workers. It's likely a problem with networkin on Docker Compose.

We strongly discurage using docker-compose for production unless you know in detail what you are doing and can diagnose and debug docker-compose issues like that yourself. There are many ways misconfigured docker compose can go wrong. We have multiple examples of that. We encourage our users to move to kubernetes and help chart instead. Errors like that are the very reason for that.

See: https://airflow.apache.org/docs/apache-airflow/stable/start/docker.html#production-readiness

> DO NOT expect the Docker Compose below will be enough to run production-ready Docker Compose Airflow installation using it. This is truly quick-start docker-compose for you to get Airflow up and running locally and get your hands dirty with Airflow. Configuring a Docker-Compose installation that is ready for production requires an intrinsic knowledge of Docker Compose, a lot of customization and possibly even writing the Docker Compose file that will suit your needs from the scratch. It’s probably OK if you want to run Docker Compose-based deployment, but short of becoming a Docker Compose expert, it’s highly unlikely you will get robust deployment with it.

> If you want to get an easy to configure Docker-based deployment that Airflow Community develops, supports and can provide support with deployment, you should consider using Kubernetes and deploying Airflow using [Official Airflow Community Helm Chart](https://airflow.apache.org/docs/helm-chart/stable/index.html).
